### ISSUE TITLE
Problem with `local_files_only`

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/diffusers#1970]
### Describe the bug

There is a bug in `DiffusionPIpeline.from_pretrained()`.

I run `from_pretrained('repo-id')` on a machine with internet connection, then disable the internet and run from_pretrained() with the option local_files_only=True. However, the call fails with a ProxyError('Cannot connect to proxy')

When I try the same thing with a Transformers model, such as BertTokenizerFast, the call succeeds and uses the cached model without complaint. I can confirm that both the diffusers and the transformers models are in the cache in the expected place and format.

Here's my test case:
1. With Internet turned on:
from diffusers import DiffusionPipeline
a = DiffusionPipeline.from_pretrained('stabilityai/stable-diffusion-2-1')

2. With Internet turned off (by adding a firewall rule):
```
from diffusers import DiffusionPipeline
a = DiffusionPipeline.from_pretrained('stabilityai/stable-diffusion-2-1',local_files_only=True
```

Error report:
MaxRetryError: HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: 
/api/models/stabilityai/stable-diffusion-2-1 (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 403 
Forbidden')))

The same pattern with BertTokenizerFast works fine:
from transformers import BertTokenizerFast
a = BertTokenizerFast.from_pretrained('bert-base-uncased',local_files_only=True)

### Reproduction

See above

### System Info

Main

**Comment by n00mkrad** [comment]
Having the same issue with `StableDiffusionInstructPix2PixPipeline`

**Comment by patrickvonplaten** [comment]
Should be fixed with #2119 :-)
